wallet/src/main/scala/org/bitcoins/wallet/internal/UtxoHandling.scala

```diff
@@ -96,59 +96,78 @@ private[wallet] trait UtxoHandling extends WalletLogger {
       spendingInfoDbs: Vector[SpendingInfoDb]): Future[
     Vector[SpendingInfoDb]] = {
 
-    val byBlock = spendingInfoDbs.groupBy(_.blockHash)
-
-    val toUpdateFs = byBlock.map {
-      case (Some(blockHash), txos) =>
-        chainQueryApi.getNumberOfConfirmations(blockHash).map {
-          case None =>
-            logger.warn(
-              s"Given txos exist in block (${blockHash.hex}) that we do not have or that has been reorged! $txos")
-            Vector.empty
-          case Some(confs) =>
-            txos.flatMap { txo =>
-              txo.state match {
-                case TxoState.ImmatureCoinbase =>
-                  if (confs > Consensus.coinbaseMaturity) {
+    val txIds =
+      spendingInfoDbs.map(_.txid) ++ spendingInfoDbs.flatMap(_.spendingTxIdOpt)
+
+    val byBlockF = {
+      transactionDAO.findByTxIdBEs(txIds).map { txDbs =>
+        val blockHashMap = txDbs.map(db => db.txIdBE -> db.blockHashOpt).toMap
+        val blockHashAndDb = spendingInfoDbs.map { txo =>
+          val txToUse = txo.state match {
+            case _: ReceivedState | DoesNotExist | ImmatureCoinbase =>
```

Comment:
is this correct in the case that we receive and a spend a transaction the transaction in the same block? I.e. spending an unconfirmed transaction? 

Wouldn't this be an issue with CPFP?

Reply:
That would be fine, the parent tx would be in the `ReceivedState` as a `PendingConfirmationReceived`

Reply:
actually realizing now `Reserved` in handled incorrectly here because it is a `SpentState`